### PR TITLE
S3C-349 backbeat routes

### PR DESCRIPTION
**Pull request by jonathan-gramain**
## Description

Add internal routes to allow backbeat to put raw data and metadata on destination

### Motivation and context

part of backbeat replication extension

### Related issues

https://scality.atlassian.net/browse/S3C-349

### CODE REVIEW

File: lib/services.js [hunk 1]
```diff
@@ -115,7 +116,7 @@ export default {
         // to return the exact date in the response
         omVal['last-modified'] = lastModifiedDate || new Date().toJSON();
         omVal['content-md5'] = contentMD5;
-
+        omVal['x-amz-replication-status'] = replication;
```
Comment (bennettbuchanan) [inline lib/services.js]: I guess this is fine for this PR, but note that I am anticipating that the header `x-amz-replication-status` will not be used, and instead a `replication` property will be added to the object metadata.

File: lib/routes/routeBackbeat.js [hunk 2]
```diff
@@ -0,0 +1,186 @@
...
+        let omVal;
+        try {
+            omVal = JSON.parse(payload);
+        } catch (err) {
+            // FIXME: add error type MalformedJSON
+            return callback(errors.MalformedPOSTRequest);
+        }
+        // specify both 'versioning' and 'versionId' to create a "new"
+        // version (updating master as well) but with specified
+        // versionId
```
Comment (electrachong) [inline lib/routes/routeBackbeat.js]: afaik this shouldn't work with real metadata/arsenal, the versioning request processor will return `BadRequest`:
https://github.com/scality/Arsenal/blob/master/lib/versioning/VersioningRequestProcessor.js#L264

Were you planning to modify the vrp code to accomplish what you want?

Comment (jonathan-gramain) [inline lib/routes/routeBackbeat.js]: Yes (as discussed offline): https://github.com/scality/Arsenal/pull/262

**Comment by jonathan-gramain** [comment]
@ironman-machine try

File: lib/routes/routeBackbeat.js [hunk 3]
```diff
@@ -0,0 +1,198 @@
...
+            objectKey: request.objectKey, resourceType: request.resourceType,
+        });
+        return responseJSONBody(errors.MethodNotAllowed, null, response, log);
+    }
+    const requestContexts = prepareRequestContexts('objectReplicate',
+                                                   request);
+    return async.waterfall([next => auth.server.doAuth(
+        request, log, (err, userInfo) => {
+            if (err) {
+                log.info('authentication error',
```
Comment (rahulreddy) [inline lib/routes/routeBackbeat.js]: This should be a `debug` level log.

**Comment by jonathan-gramain** [comment]
@ironman-machine try

**Pull request merged**